cmake_test/add_dir.cmake

```diff
@@ -0,0 +1,33 @@
+include_guard()
+set(_ct_add_dir_file_dir "${CMAKE_CURRENT_LIST_DIR}")
```

Comment:
If you put it here, you grab the directory where this file is included. If you put it in the function you grab the directory where the function is being executed. In general they're not the same and I'd assume you want the latter.

Reply:
Actually I'm using this to resolve the location of the templates, determining that I found was quite difficult and this was the best solution I could come up with. Do you happen to have a better solution than this?

Reply:
whoops. My bad. I guess I didn't understand how you were using that variable. AFAIK for getting the directory of the templates that's the only way (aside from hard-coding/assuming it). You can pretty it up by declaring a function `get_template_dir` or something, but fundamentally you're still going to need a `set` call like what you have.